tests/unit/test_relational_config.py

```diff
@@ -0,0 +1,114 @@
+#!/usr/bin/env python3
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+"""Feature: Relation-dependant Opentelemetry-collector config.
+
+Scenario: Relate to a data sink
+    Given   A valid config requires at least one pipeline and it must have a at least one
+            receiver-exporter pair
+    And     An OTLP receiver is added to all signal (logs, metrics, traces) pipelines
+    When    A data-sink charm is integrated with the otelcol charm
+    Then    A relevant exporter is added to the config
+
+Scenario: Relate to a data source
+    Given   A valid config requires at least one pipeline and it must have a at least one
+            receiver-exporter pair
+    When    A data-source charm is integrated with the otelcol charm
+    Then    A relevant receiver is added to the config
+    And     If there is no exporter for that pipeline
+    Then    A debug exporter is added for that receiver-exporter pair to ensure a valid config
+"""
```

Comment:
Repeating the BDD statements here is similar to having a `*.feature` file: we now have 2 separate place we'd need to keep up to date.

Wdyt about:
- Module level docstring = "Feature:" statement
- Testfunc docstring = "Scenario:" statement
- Inline "loose-bdd" comments